include/mvdtool/bits/mvd3_misc.hpp

```diff
@@ -220,175 +183,204 @@ inline bool MVD3File::hasRotations() const {
 }
 
-
-inline std::vector<std::string> MVD3File::listAllEtypes() const{
-    const auto& getEtypesLambda = [=](std::vector<std::string>& me_combos, std::vector<std::string>& morphologies){return _tsv_file->getEtypes(me_combos, morphologies);};
-    auto allEtypes = getDataFromTsvOrMvd3WithIndex<std::string>(did_lib_data_etypes, did_cells_index_etypes, Range(0,0), getEtypesLambda);
-    vector_remove_dups(allEtypes);
-    return allEtypes;
+inline std::vector<std::string> MVD3File::listAllEtypes() const {
+    return getDataFromMVD(did_lib_data_etypes, did_lib_NONE, range_ALL);
 }
 
-inline std::vector<std::string> MVD3File::listAllMtypes() const{
-    const auto& getMtypesLambda = [=](std::vector<std::string>& me_combos, std::vector<std::string>& morphologies){return _tsv_file->getMtypes(me_combos, morphologies);};
-    auto allMtypes = getDataFromTsvOrMvd3WithIndex<std::string>(did_lib_data_mtypes, did_cells_index_mtypes, Range(0,0), getMtypesLambda);
-    vector_remove_dups(allMtypes);
-    return allMtypes;
+inline std::vector<std::string> MVD3File::listAllMtypes() const {
+    return getDataFromMVD(did_lib_data_mtypes, did_lib_NONE, range_ALL);
 }
 
-inline std::vector<boost::int32_t> MVD3File::listAllLayers() const{
-    const auto& getLayersLambda = [=](std::vector<std::string>& me_combos, std::vector<std::string>& morphologies){return _tsv_file->getLayers(me_combos, morphologies);};
-    auto allLayers = getDataFromTsvOrMvd3WithoutIndex<boost::int32_t>(did_cells_layer, Range(0,0), getLayersLambda);
-    vector_remove_dups(allLayers);
-    return allLayers;
+inline std::vector<std::string> MVD3File::listAllRegions() const {
+    return getDataFromMVD(did_lib_data_regions, did_lib_NONE, range_ALL);
 }
 
-inline std::vector<std::string> MVD3File::listAllEmodels() const{
-    const auto& getEmodelsLambda = [=](std::vector<std::string>& me_combos, std::vector<std::string>& morphologies){return _tsv_file->getEmodels(me_combos, morphologies);};
-    auto allEmodels = getDataFromTsv<std::string>("eModel", Range(0,0), getEmodelsLambda);
-    vector_remove_dups(allEmodels);
-    return allEmodels;
+inline std::vector<std::string> MVD3File::listAllSynapseClass() const {
+    return getDataFromMVD(did_lib_data_syn_class, did_lib_NONE, range_ALL);
 }
 
-inline std::vector<std::string> MVD3File::listAllRegions() const{
-    HighFive::DataSet index = _hdf5_file.getDataSet(did_lib_data_regions);
-    return get_data_for_selection<std::string>(index, Range(0,0));
+// Emodels are only avail within TSV
+inline std::vector<std::string> MVD3File::listAllEmodels() const {
+    if(!_tsv_file) {
+        throw MVDException("No TSV file is opened with MVD3 to extract all the emodels.");
+    }
+    std::vector<std::string> emodels;
+    const auto all_entries = _tsv_file->getAll();
+    emodels.reserve(all_entries.size());
+    for (const auto& entry : all_entries) {
+        emodels.push_back(entry.get().eModel);
+    }
+    return emodels;
```

Comment:
@ferdonline As far as I understood `listAll<Fields>` functions return unique values. In this case I think that the `emodels` array should be filtered to remove duplicate values, since there could be duplicate values coming in the `tsv` file.

Reply:
That's a good point. I woudn't expect emodels to be repeated within the tsv, but who knows... Thanks!